src/libaktualizr/primary/aktualizr.cc

```diff
@@ -170,13 +170,30 @@ boost::signals2::connection Aktualizr::SetSignalHandler(
   return sig_->connect(handler);
 }
 
-std::unique_ptr<StorageTargetRHandle> Aktualizr::GetStoredTarget(const Uptane::Target &target) {
+std::vector<Uptane::Target> Aktualizr::GetStoredTargets() {
+  try {
+    return storage_->getTargetFiles();
+  } catch (...) {
+    return {};
+  }
+}
+
+bool Aktualizr::DeleteStoredTarget(const Uptane::Target &target) {
+  try {
+    storage_->removeTargetFile(target.filename());
+  } catch (...) {
```

Comment:
Probably we should discuss that again, but haven't we decided to let exceptions propagate through the C++ API level? That's also what other API calls do.